### ISSUE TITLE
Mod Request: Recipe Cache

### ISSUE THREAD

**Issue by mr-pi12** [ReperakPro/fabric-optimization-mods#38]
**Requirements for a mod to be added:**
1. The mod must be open-source.
2. The mod cannot sacrifice vanilla behavior for performance.
3. The mod must be actively maintained.

**Mod name:** Recipe Cache

**Type of mod (e.g. rendering, lighting, extra):** ticking optimization (crafting tables and furnaces)

**Repository (e.g. GitHub, GitLab, Gitea):** https://gitlab.com/biom4st3r/optimizedcrafting

**Developer/organization:** biome4st3r

**Additional notes:**
This mod seems like it does the same as fastbench and fastfurnace, but is actively maintained. It also has less bugs from fastfurnace, and acts as both mods in one. However, I haven't tested this mod extensively myself, so I don't know if it offers significant performance benefits on servers. I also found a bug myself in fastbench - using it on an online server made the items I crafted invisible - this isn't in recipecache, so perhaps it's a better alternative?

**Comment by ReperakPro** [comment]
I actually took a look at that mod last night, but when I looked at the code it was really poorly made. The code is entirely Mixins with no helpers or anything. I'm leaving this open as I can't confirm nor deny that this has any performance benefits, so if possible, I'd like to see some numbers from either the creator or any user.

**Comment by mr-pi12** [comment]
Yeah, I didn't have a lot of basis for the performance benfits of this mod, so I didn't know whether it would be classed as an optimizarion mod.

**Comment by ReperakPro** [comment]
The developer even left a remapped jar in their GitLab repo :joy:

**Comment by mr-pi12** [comment]
I used this mod, and it seems to reduce networking overhead on servers and there seem to be fewer fps drops in singleplayer when opening a crafting table or furnace over and over (on my pc anyway - which isn't a great benchmark, seeing as the parts are nearing to a decade old).
